timelock-agent/src/test/java/com/palantir/timelock/config/PaxosInstallConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.config;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class PaxosInstallConfigurationIntegrationTest {
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private File fileSubdirectory;
+    private File sqliteSubdirectory;
+
+    @Before
+    public void setUp() {
+        fileSubdirectory = temporaryFolder.getRoot().toPath().resolve("file").toFile();
+        sqliteSubdirectory = temporaryFolder.getRoot().toPath().resolve("sqlite").toFile();
+    }
+
+    @Test
+    public void serviceIsNewIfNoDataDirectoriesPresent() {
+        ImmutablePaxosInstallConfiguration.Builder partialConfiguration = createPartialConfiguration(
+                fileSubdirectory, sqliteSubdirectory);
+        assertThatCode(() -> partialConfiguration.isNewService(true).build()).doesNotThrowAnyException();
+        assertThatThrownBy(() -> partialConfiguration.isNewService(false).build())
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("The timelock data directories do not appear to exist.");
```

Comment:
I't kinda funky that this is not an IllegalStateException tbh

Reply:
Yep, I'd have used `State` if fresh, but preserving legacy behaviour.